sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -1516,9 +1516,10 @@ class CudfRegexTranspiler(mode: RegexMode) {
         case (RegexRepetition(_, SimpleQuantifier('*')), SimpleQuantifier('+')) =>
           throw new RegexUnsupportedException("Possessive quantifier *+ not supported",
             quantifier.position)
-        case (RegexRepetition(_, SimpleQuantifier('*')), SimpleQuantifier('?')) =>
-          throw new RegexUnsupportedException("Lazy quantifier *? not supported",
-            quantifier.position)
+        case (RegexRepetition(_, SimpleQuantifier('?')) |
+              RegexRepetition(_, SimpleQuantifier('*')) |
+              RegexRepetition(_, SimpleQuantifier('+')), SimpleQuantifier('?')) =>
```

Comment:
nit: the pattern matching can be more concise here

```suggestion
        case (RegexRepetition(_, SimpleQuantifier('?' | '*' | '+')), SimpleQuantifier('?')) =>
```